mdoc/src/main/scala-3/mdoc/internal/markdown/Instrumenter.scala

```diff
@@ -2,105 +2,170 @@ package mdoc.internal.markdown
 
 import java.io.ByteArrayOutputStream
 import java.io.PrintStream
-import java.nio.file.Path
-
-import dotty.tools.dotc.interfaces.SourcePosition
-import dotty.tools.dotc.ast.untpd._
-import dotty.tools.dotc.core.Contexts.Context
-import dotty.tools.dotc.ast.Trees
-import dotty.tools.dotc.core.Flags
-
+import scala.meta._
+import scala.meta.inputs.Position
+import Instrumenter.position
+import mdoc.internal.markdown.Instrumenter.Binders
+import scala.meta.Mod.Lazy
 import scala.collection.mutable
 import mdoc.Reporter
 import mdoc.internal.cli.InputFile
+import java.nio.file.Path
 import mdoc.internal.cli.Settings
-import scala.meta.Mod.Lazy
-import scala.meta.Name
 
-import Instrumenter.position
+class CodePrinter(ps: PrintStream, baseIndent: Int = 0, baseNest: Int = 0) {
```

Comment:
The story here is that I overwrote (accidentally) the old version because it stopped being picked up.

This code printer seems to be handling all cases where correct indentation is needed, but I split it out to scala 3 sources to avoid changing the mainline scala 2 behaviour.